livre/core/pipeline/PortType.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _PortType_h_
+#define _PortType_h_
+
+#include <livre/core/types.h>
+
+namespace livre
+{
+
+/**
+ * PortType class is base class for keeping the track for types
+ * by using the std::type_index.
+ */
+class PortType
```

Comment:
If this class really needed?
I think it's preferable to duplicate the getDataType() member function in PortInfo and PortData and remove the inheritance for two reasons. Inheritance seems overkill to refactor just a member variable and a member function and I don't think it makes the code any more readable. As a user I have to look up what PortType is for a very small thing. To give you an example, this would be like inheriting from has_size for any class that needs a size() method in std. The only reason for doing such a thing would be concept checking in template metaprogramming.